azurerm/resource_arm_eventhub_namespace_disaster_recovery_config.go

```diff
@@ -0,0 +1,319 @@
+			return fmt.Errorf("Error waiting for replication to compelte for EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", name, namespaceName, resourceGroup, err)
+		}
+	}
+	read, err := client.Get(ctx, resourceGroup, namespaceName, name)
+	if err != nil {
+		return fmt.Errorf("Error reading EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %v", name, namespaceName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Got nil ID for EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q)", name, namespaceName, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmEventHubNamespaceDisasterRecoveryConfigRead(d, meta)
+}
+
+func resourceArmEventHubNamespaceDisasterRecoveryConfigUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).eventhub.DisasterRecoveryConfigsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	name := id.Path["disasterRecoveryConfigs"]
+	resourceGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+
+	if d.HasChange("partner_namespace_id") {
+
+		// break pairing
+		breakPair, err := client.BreakPairing(ctx, resourceGroup, namespaceName, name)
+		if breakPair.StatusCode != http.StatusOK {
+			return fmt.Errorf("Error issuing break pairing request for EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", name, namespaceName, resourceGroup, err)
+		}
+
+		if err := resourceArmEventHubNamespaceDisasterRecoveryConfigWaitForState(ctx, client, resourceGroup, namespaceName, name); err != nil {
+			return fmt.Errorf("Error waiting for break pairing request to compelte for EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", name, namespaceName, resourceGroup, err)
+		}
+
+	}
+
+	parameters := eventhub.ArmDisasterRecovery{
+		ArmDisasterRecoveryProperties: &eventhub.ArmDisasterRecoveryProperties{
+			PartnerNamespace: utils.String(d.Get("partner_namespace_id").(string)),
+		},
+	}
+
+	if v, ok := d.GetOk("alternate_name"); ok {
+		parameters.ArmDisasterRecoveryProperties.AlternateName = utils.String(v.(string))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, namespaceName, name, parameters); err != nil {
+		return fmt.Errorf("Error creating/updating EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", name, namespaceName, resourceGroup, err)
+	}
+
+	if v, ok := d.GetOkExists("wait_for_replication"); ok && v.(bool) {
+		if err := resourceArmEventHubNamespaceDisasterRecoveryConfigWaitForState(ctx, client, resourceGroup, namespaceName, name); err != nil {
+			return fmt.Errorf("Error waiting for replication to compelte for EventHub Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", name, namespaceName, resourceGroup, err)
+		}
+	}
```

Comment:
(as above) we should wait for this to be replicated before returning - so can we remove this `if` statement here